homeassistant/components/light/vera.py

```diff
@@ -22,62 +21,23 @@
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup Vera lights."""
-    import pyvera as veraApi
-
-    base_url = config.get('vera_controller_url')
-    if not base_url:
-        _LOGGER.error(
-            "The required parameter 'vera_controller_url'"
-            " was not found in config"
-        )
-        return False
-
-    device_data = config.get('device_data', {})
-
-    vera_controller, created = veraApi.init_controller(base_url)
-
-    if created:
-        def stop_subscription(event):
-            """Shutdown Vera subscriptions and subscription thread on exit."""
-            _LOGGER.info("Shutting down subscriptions.")
-            vera_controller.stop()
-
-        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_subscription)
-
-    devices = []
-    try:
-        devices = vera_controller.get_devices([
-            'Switch',
-            'On/Off Switch',
-            'Dimmable Switch'])
-    except RequestException:
-        # There was a network related error connecting to the vera controller.
-        _LOGGER.exception("Error communicating with Vera API")
-        return False
-
-    lights = []
-    for device in devices:
-        extra_data = device_data.get(device.device_id, {})
-        exclude = extra_data.get('exclude', False)
-
-        if exclude is not True:
-            lights.append(VeraLight(device, vera_controller, extra_data))
+    vera = get_component('vera')
+    devices = vera.VERA_DEVICES[vera.LIGHT]
+    controller = vera.VERA_CONTROLLER
 
+    lights = [VeraLight(device, controller)
+              for device in devices]
     add_devices_callback(lights)
```

Comment:
btw, I know you didn't write the bottom of this file but in the light class, there is no need to use `STATE_ON`, `STATE_OFF`. You merely want to store a boolean in `update()` and return that in `is_on()`.